pkg/webhooks/prometheusrule/prometheusrule_test.go

```diff
@@ -0,0 +1,306 @@
+			testID:          "admin-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-monitoring",
+			username:        "kube:admin",
+			targetResource:  "prometheusrule",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Create,
+			shouldBeAllowed: true,
+		},
+		{
+
+			testID:          "admin-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-monitoring",
+			username:        "kube:admin",
+			targetResource:  "prometheusrule",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Update,
+			shouldBeAllowed: true,
+		},
+		{
+
+			testID:          "admin-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-monitoring",
+			username:        "kube:admin",
+			targetResource:  "prometheusrule",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Delete,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-create-prometheusrule-in-openshift-user-workload-monitoring",
+			targetNamespace: "openshift-user-workload-monitoring",
+			targetResource:  "prometheusrule",
+			username:        "user1",
+			userGroups:      []string{"cluster-admins", "system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Create,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-create-prometheusrule-in-openshift-user-workload-monitoring",
+			targetNamespace: "openshift-user-workload-monitoring",
+			targetResource:  "prometheusrule",
+			username:        "user2",
+			userGroups:      []string{"cluster-admins", "system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Delete,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-create-prometheusrule-in-openshift-user-workload-monitoring",
+			targetNamespace: "openshift-user-workload-monitoring",
+			targetResource:  "prometheusrule",
+			username:        "user3",
+			userGroups:      []string{"cluster-admins", "system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Update,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-create-prometheusrule-in-redhat-rhoam-observability",
+			targetNamespace: "redhat-rhoam-observability",
+			targetResource:  "prometheusrule",
+			username:        "system:serviceaccounts:redhat-ns",
+			userGroups:      []string{"system:serviceaccounts:redhat-ns", "system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Create,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-update-prometheusrule-in-redhat-rhoam-observability",
+			targetNamespace: "redhat-rhoam-observability",
+			targetResource:  "prometheusrule",
+			username:        "system:serviceaccounts:redhat-ns",
```

Comment:
username should be something like "system:serviceaccounts:redhat-ns:test-operator"